src/stactools/core/utils/raster_footprint.py

```diff
@@ -18,6 +21,622 @@
+            )
+        if len(set(reader.nodatavals)) != 1:
+            raise ValueError("All raster bands must have the same 'nodata' value.")
+
+        if not bands:
+            bands = reader.indexes
+        if no_data is None:
+            no_data = reader.nodata
+
+        band_data = []
+        for index in bands:
+            band_data.append(reader.read(index))
+
+        return cls(
+            data_array=np.asarray(band_data),
+            crs=reader.crs,
+            transform=reader.transform,
+            no_data=no_data,
+            precision=precision,
+            densification_factor=densification_factor,
+            densification_distance=densification_distance,
+            simplify_tolerance=simplify_tolerance,
+        )
+
+    @classmethod
+    def update_geometry_from_asset_footprint(
+        cls,
+        item: Item,
+        *,
+        asset_names: List[str] = [],
+        precision: int = DEFAULT_PRECISION,
+        densification_factor: Optional[int] = None,
+        densification_distance: Optional[float] = None,
+        simplify_tolerance: Optional[float] = None,
+        no_data: Optional[Union[int, float]] = None,
+        bands: List[int] = [1],
+        skip_errors: bool = True,
+    ) -> bool:
+        """
+        Accepts an Item and an optional list of asset names within that Item, and
+        updates the geometry of that Item in-place with the data footprint derived
+        from the first of the assets that exists in the Item.
+
+        See :class:`RasterFootprint` for details on the data footprint
+        calculation.
+
+        Args:
+            item (Item): The PySTAC Item to update.
+            asset_names (List[str]): The names of the assets for which to attempt to
+                extract footprints. The first successful footprint will be used. If
+                the list is empty, all assets will be tried until one is successful.
+            precision (int): The number of decimal places to include in the final
+                footprint coordinates.
+            densification_factor (Optional[int]): The factor by which to
+                increase point density within the polygon before projection to
+                EPSG 4326. A factor of 2 would double the density of points (placing
+                one new point between each existing pair of points), a factor of
+                3 would place two points between each point, etc. Higher
+                densities produce higher fidelity footprints in areas of high
+                projection distortion.
+            simplify_tolerance (Optional[float]): Distance, in degrees, within which
+                all locations on the simplified polygon will be to the original
+                polygon.
+            no_data (Optional[Union[int, float]]): Explicitly sets the nodata value
+                if not in source image metadata. If set to None, this will return
+                a footprint including nodata values.
```

Comment:
This is missing `densification_distance`, which tells me that we're repeating these docs in too many places. I think it could be appropriate to enumerate the full meaning once (e.g. in the class definition) and just point to those docs with references.

Reply:
Good catch. I agree that it is a lot of repetition. Unfortunately the class definition does not contain all arguments, as some of them are only specified in the alternative constructors. And the alternative constructors themselves differ slightly in their arguments.

For now, I've pasted in the missing argument. I also like being able to see the full list of argument definitions via "intellisense", so I'd lean toward keeping the repetitive lists.